File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        startIdx=0
        endIdx=len(height)-1
        area=0
        while endIdx>startIdx:
            area=max(area,min(height[endIdx],height[startIdx]) * (endIdx-startIdx))
            if height[endIdx] > height[startIdx]:
                startIdx+=1
            else:
                endIdx-=1
        return area

File: 11-container-with-most-water/container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let startIdx = 0;
    let endIdx = (height.length) -1;
    let area = 0;
    while(endIdx>startIdx){
        area = Math.max(area, Math.min(height[endIdx],height[startIdx]) *(endIdx-startIdx));
        if(height[endIdx] > height[startIdx]){
            startIdx+=1;
        }
        else{
            endIdx-=1;
        }
    };
    return area;
};
